File: Front/RHStoreWS/RHStoreUsuariosBO/CuponBO.cs
using RHStoreBaseBO;
using RHStoreBaseBO.ServiciosWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RHStoreUsuariosBO
{
	public class CuponBO : BaseBO
	{
        public int insertar(string codigo, string descripcion, double valorDescuento, DateTime fechaInicio, DateTime fechaFin, trabajador _trabajador)
        {
            return CuponWS.insertarCupon(codigo, descripcion, valorDescuento, fechaInicio, fechaFin, _trabajador);
        }

        public int modificar(int idCupon, string codigo, string descripcion, double valorDescuento, DateTime fechaInicio, DateTime fechaFin, trabajador _trabajador)
        {
            return CuponWS.modificarCupon(idCupon, codigo, descripcion, valorDescuento, fechaInicio, fechaFin, _trabajador);
        }

        public int eliminar(int idCupon)
        {
            return CuponWS.eliminarCupon(idCupon);
        }

        public BindingList<cupon> listarTodos()
        {
            return new BindingList<cupon>(CuponWS.listarCuponesTodos());
        }

        public cupon obtenerPorId(int idCupon)
        {
            return CuponWS.obtenerCuponPorId(idCupon);
        }

		public BindingList<cupon> listarPorCodigoDescripcion(string cadena)
		{
			return new BindingList<cupon>(CuponWS.listarCuponesPorCodigoDescripcion(cadena));
		}
	}
}

File: Back/SoftRhUsuariosBO/src/pe/edu/pucp/softrh/usuarios/bo/CuponBO.java
package pe.edu.pucp.softrh.usuarios.bo;

import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softrh.usuarios.dao.CuponDAO;
import pe.edu.pucp.softrh.usuarios.daoimp.CuponDAOImp;
import pe.edu.pucp.softrh.usuarios.model.Cupon;
import pe.edu.pucp.softrh.usuarios.model.Trabajador;

public class CuponBO {
	private CuponDAO cuponDAO;

	public CuponBO() {
		this.cuponDAO = new CuponDAOImp();
	}

	public Integer insertar(String codigo, String descripcion, Double valorDescuento, Date fechaInicio, Date fechaFin, Trabajador trabajador) {
		Cupon cupon = new Cupon(codigo, descripcion, valorDescuento, fechaInicio, fechaFin, trabajador);
		cupon.setIdCupon(cuponDAO.insertar(cupon));
		return cupon.getIdCupon();
	}

	public Integer modificar(Integer idCupon, String codigo, String descripcion, Double valorDescuento, Date fechaInicio, Date fechaFin, Trabajador trabajador) {
		Cupon cupon = new Cupon(codigo, descripcion, valorDescuento, fechaInicio, fechaFin, trabajador);
		cupon.setIdCupon(idCupon);
		return cuponDAO.modificar(cupon);
	}

	public Integer eliminar(Integer idCupon) {
		return cuponDAO.eliminar(idCupon);
	}

	public ArrayList<Cupon> listarTodos() {
		return cuponDAO.listarTodos();
	}

	public Cupon obtenerPorId(Integer idCupon) {
		return cuponDAO.obtenerPorId(idCupon);
	}

	public ArrayList<Cupon> listarPorCodigoDescripcion(String cadena) {
		return cuponDAO.listarPorCodigoDescripcion(cadena);
	}
}
